core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -560,7 +560,13 @@ class ContainerProxy(
         }
         val parameters = job.msg.content getOrElse JsObject.empty
 
-        val authEnvironment = job.msg.user.authkey.toEnvironment
+        // if the action requests the api key to be injected into the action context, add it here;
+        // treat a missing annotation as requesting the api key for backward compatibility
+        val authEnvironment = {
+          if (job.action.annotations.isTruthy(WhiskAction.provideApiKeyAnnotationName, valueForNonExistent = true)) {
+            job.msg.user.authkey.toEnvironment
+          } else JsObject.empty
```

Comment:
```
} else job.msg.user.authkey.toEnvironmentSensitive
```

I do suggest to move the decision what to hide as sensitive information to each `GenericAuthKey` implementation 
by adding a new method `authkey.toEnvironmentSensitive`. This method should return `JsObject.empty`
for the BasicAuthenticationAuthKey. Other `GenericAuthKey` implementations could decide which of the information stored should be made available on request only. 

If we stay backwards compatible , I would be willing to implement this change implement  in a following PR:.

Reply:
👍

Reply:
Can you send a comment to the dev list with the proposal. It is reasonable to me. Do you have an example you can attach?

Reply:
I will address it once this PR is merged. No need to delay.

Reply:
Thanks!